core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -376,19 +408,16 @@ private void applyConfiguration(CreateContainerCmd createCommand) {
             String alias = linkEntries.getKey();
             LinkableContainer linkableContainer = linkEntries.getValue();
 
-            Set<Link> links = dockerClient.listContainersCmd().exec().stream()
-                    .filter(container -> container.getNames()[0].endsWith(linkableContainer.getContainerName()))
-                    .map(container -> new Link(container.getNames()[0], alias))
+            Set<Link> links = dockerClient.listContainersCmd()
```

Comment:
I would like this in a private method as well (since I really have to look into the stream to get whats happening here). Maybe `findLinksForX` (whatever X is)?